src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2411,49 +2410,53 @@ impl Coordinator {
                 timeout_dur = Duration::MAX;
             }
 
-            let make_diffs = move |rows: Vec<Row>| -> Result<Vec<(Row, Diff)>, AdapterError> {
-                let arena = RowArena::new();
-                // Use 2x row len incase there's some assignments.
-                let mut diffs = Vec::with_capacity(rows.len() * 2);
-                let mut datum_vec = mz_repr::DatumVec::new();
-                for row in rows {
-                    if !assignments.is_empty() {
-                        assert!(
-                            matches!(kind, MutationKind::Update),
-                            "only updates support assignments"
-                        );
-                        let mut datums = datum_vec.borrow_with(&row);
-                        let mut updates = vec![];
-                        for (idx, expr) in &assignments {
-                            let updated = match expr.eval(&datums, &arena) {
-                                Ok(updated) => updated,
-                                Err(e) => return Err(AdapterError::Unstructured(anyhow!(e))),
-                            };
-                            updates.push((*idx, updated));
+            let make_diffs =
+                move |mut rows: Box<dyn RowIterator>| -> Result<Vec<(Row, Diff)>, AdapterError> {
+                    let arena = RowArena::new();
+                    // Use 2x row len incase there's some assignments.
```

Comment:
This comment should go away if we're not using with_capacity.

Reply:
removed!